packages/mongoose/src/index.ts

```diff
@@ -0,0 +1,146 @@
+    "VerificationToken",
+    verificationTokenSchema
+  )
+
+  return {
+    async createUser(data) {
+      const user = await UserModel.create(data)
+      return from<AdapterUser>(user)
+    },
+    async getUser(id) {
+      try {
+        const user = await UserModel.findById(id).lean()
+        if (!user) return null
+        return from<AdapterUser>(user)
+      } catch (err) {
+        return null
+      }
+    },
+    async getUserByEmail(email) {
+      const user = await UserModel.findOne({ email: email }).lean()
+      if (!user) return null
+      return from<AdapterUser>(user)
+    },
+    async getUserByAccount(data) {
+      const account = await AccountModel.findOne(data)
+      if (!account) return null
+      const user = await UserModel.findById(account.userId).lean()
+      if (!user) return null
+      return from<AdapterUser>(user)
+    },
+    async updateUser(data) {
+      const user = await UserModel.findByIdAndUpdate(
+        data.id,
+        { name: data.name },
+        { new: true }
+      ).exec()
+      return from<AdapterUser>(user)
+    },
+    async deleteUser(id) {
+      await Promise.all([
+        AccountModel.deleteMany({ userId: id }),
+        SessionModel.deleteMany({ userId: id }),
+        UserModel.findByIdAndDelete(id),
+      ])
+    },
+    async linkAccount(data) {
+      const account = await AccountModel.create(data)
+      return from<Account>(account)
+    },
+    async unlinkAccount(data) {
+      const account = await AccountModel.findOneAndDelete(data)
+      return from<Account>(account)
+    },
+    async getSessionAndUser(sessionToken) {
+      const session = await SessionModel.findOne({
+        sessionToken: sessionToken,
+      }).lean()
+      if (!session) return null
+      const user = await UserModel.findById(session.userId).lean()
+      if (!user) return null
+      return {
+        user: from<AdapterUser>(user),
+        session: from<AdapterSession>(session),
+      }
+    },
+    async createSession(data) {
+      const session = await SessionModel.create(data)
+      return from<AdapterSession>(session)
+    },
+    async updateSession(data) {
+      const session = await SessionModel.findOneAndUpdate({
+        sessionToken: data.sessionToken,
+        expires: data.expires,
+      })
+      return from<AdapterSession>(session)
+    },
+    async deleteSession(sessionToken) {
+      const session = await SessionModel.findOneAndDelete({
+        sessionToken: sessionToken,
+      })
+      return from<AdapterSession>(session)
+    },
+    async createVerificationToken(data) {
+      const verificationToken = await VerificationTokenModel.create(data)
+      return from<VerificationToken>(verificationToken)
+    },
+    async useVerificationToken(data) {
+      const verificationToken = await VerificationTokenModel.findOneAndDelete(
+        data
+      ).lean()
+      if (!verificationToken) return null
+      const { _id, __v, ...rest } = verificationToken
```

Comment:
Question: We can use the `from` function from `format` here right? Or else, there will be 2 unused variables